common/static/common/js/components/views/tabbed_view.js

```diff
@@ -89,33 +110,128 @@
 
-                   setActiveTab: function (index) {
+                   setActiveTab: function(index) {
                        var tabMeta = this.getTabMeta(index),
                            tab = tabMeta.tab,
                            tabEl = tabMeta.element,
                            view = tab.view;
+                       
                        // Hide old tab/tabpanel
-                       this.$('button.is-active').removeClass('is-active').attr('aria-expanded', 'false');
-                       this.$('.tabpanel[aria-expanded="true"]').attr('aria-expanded', 'false').addClass('is-hidden');
+                       this.$('button.is-active')
+                        .removeClass('is-active')
+                        .attr({
+                            'aria-expanded': 'false',
+                            'aria-selected': 'false',
+                            'tabindex': '-1'
+                        });
+                       
+                       this.$('.tabpanel[aria-hidden="false"]')
+                        .addClass('is-hidden')
+                        .attr({
+                            'aria-hidden': 'true'
+                        });
+                       
                        // Show new tab/tabpanel
-                       tabEl.addClass('is-active').attr('aria-expanded', 'true');
-                       view.$el.attr('aria-expanded', 'true').removeClass('is-hidden');
-                       // This bizarre workaround makes focus work in Chrome.
-                       _.defer(function () {
-                           view.$('.sr-is-focusable.' + getTabPanelId(tab.url)).focus();
-                       });
                        if (this.router) {
-                           this.router.navigate(tab.url, {replace: true});
+                           this.router.navigate(tab.url, { replace: true });
                        }
+
+                       tabEl
+                        .addClass('is-active')
+                        .attr({
+                            'aria-expanded': 'true',
+                            'aria-selected': 'true',
+                            'tabindex': '0'
+                        });
+                       
+                       view.$el
+                        .removeClass('is-hidden')
+                        .attr({
+                            'aria-hidden': 'false',
+                        })
+                        .focus();
                    },
 
-                   switchTab: function (event) {
+                   switchTab: function(event) {
                        event.preventDefault();
                        this.setActiveTab($(event.currentTarget).data('index'));
                    },
+                   
+                   previousTab: function(focused, index, total, event) {
+                       event.preventDefault();
+
+                        var tab, panel;
+
+                        if (event.altKey || event.shiftKey) {
```

Comment:
IMO, these two functions would be clearer if they didn't take the ``event`` argument. They should always do what they say (e.g. go the previous tab), and the caller should not call them if certain modifiers are on.